src/openrct2-ui/windows/TopToolbar.cpp

```diff
@@ -389,6 +393,16 @@ static void window_top_toolbar_mouseup(rct_window* w, rct_widgetindex widgetInde
         case WIDX_MUTE:
             audio_toggle_all_sounds();
             break;
+        case WIDX_CHAT:
+            if (chat_available())
+            {
+                chat_toggle();
+            }
+            else
+            {
+                context_show_error(STR_CHAT_UNAVAILABLE, STR_NONE);
```

Comment:
Will this ever happen? The button is not present anyway if there is no network. I would just remove the else clause and keep the `if`.

Reply:
Whenever `network_get_status` is not 'connected' or `network_get_authstatus` is not 'auth status ok'. I suppose this could happen during a multiplayer game (e.g., connection lost).